modules/bibdk_sbkopi/includes/bibdk_sbkopi.ajax.inc

```diff
@@ -44,50 +47,119 @@ function bibdk_sbkopi_ajax_openorder_request($pid) {
+      return $delivery_method;
+    }
+    else {
+      return FALSE;
+    }
   }
   else {
     return FALSE;
   }
 }
 
 /**
- * Parse response from open order
+ * Parsing response from OpenSearch with marcxchange format.
+ * This is a temporary fix until OpenFormat can deliver single fields.
  *
- * @param $xml
  * @param $pid
- * @return bool|NULL|string
+ *
+ * @return bool|mixed|\stdClass|string
  */
-function bibdk_sbkopi_ajax_parse_response($xml, $pid) {
-  $xpath = _bibdk_sbkopi_set_xpath($xml);
-  if ($xpath === FALSE) {
-    return 'error';
-  }
-  // get articleDeliveryPossible
-  $query = '//ns:articleDeliveryPossible';
-  $nodelist = $xpath->query($query);
-  $possible = isset($nodelist->item(0)->nodeValue) ? $nodelist->item(0)->nodeValue : NULL;
-
-// get articleDirect
-  $query = '//ns:articleDirect';
-  $nodelist = $xpath->query($query);
-  $deliver = isset($nodelist->item(0)->nodeValue) ? $nodelist->item(0)->nodeValue : NULL;
-
-  if ($possible) {
-    return $deliver;
+function bibdk_sbkopi_do_marcxchangeRequest($pid) {
+  $params = array(
+    'objectFormat' => 'marcxchange',
+    'objectId' => array($pid),
+    'outputType' => 'xml',
+  );
+
+  $tc = new ting_client_class();
+  return $tc->do_request('object', $params, FALSE);
+}
+
+/**
+ * Parsing response from OpenSearch with marcxchange format.
+ * This is a temporary fix until OpenFormat can deliver single fields.
+ *
+ * @param $response
+ *
+ * @return bool|\SimpleXMLElement
+ */
+function bibdk_sbkopi_parse_marcx($response) {
+  $xml = new SimpleXMLElement(
+    $response
+  );
+
+  $namespaces = $xml->getNamespaces(TRUE);
+  foreach($namespaces AS $prefix => $namespace) {
+    $xml->registerXPathNamespace($prefix, $namespace);
   }
-  else {
-    return FALSE;
+
+  if (isset($xml->result->searchResult->collection->object)) {
+    $obj = $xml->result->searchResult->collection->object;
+
+    $record = $obj->children($namespaces['marcx'])->collection->record;
+    if (!empty($record)) {
+      $issn = "";
+
+      foreach($record->datafield AS $datafield) {
+
+        foreach ($datafield->attributes() AS $key => $value) {
+
+          $tagv = $value->__toString();
+          if ($key === "tag" && ($tagv === "022" || $tagv === "557")) {
+
+            foreach($datafield->subfield AS $subfield) {
+
+              if ($tagv === "022"
+                && $subfield->attributes()->code->__toString() === "a") {
+                $issn = $subfield->__toString();
+              }
+              else if ($tagv === "557"
+                && $subfield->attributes()->code->__toString() === "z") {
+                $issn = $subfield->__toString();
+              }
+            }
+          }
+        }
+      }
+      if (!empty($issn)) {
+        $url = variable_get('bibdk_sbkopi_journal_url');
+        $url .= '?issn=' . str_replace('-', '', $issn);
+        $url .= "&date=" . date('Y-m-d');
+        if ($content = file_get_contents($url)) {
```

Comment:
file_get_contens require php.ini allow_fopen_url to be on - it is off on most prodcution servers - use curl instead

Reply:
This has been replaced with MicroCurl.